humilis_batch/resources.yaml.j2

```
@@ -109,6 +110,17 @@ resources:
                 {% endfor %}
             {% endif %}
 
+    {% if has_launch_template %}
+    LaunchTemplate:
+      Type: "AWS::EC2::LaunchTemplate"
+      Properties:
+        LaunchTemplateName: "{{__context.environment.name}}-{{__context.layer.name}}-{{__context.stage}}"
+        LaunchTemplateData:
+          {% if user_data %}
+          UserData: {{user_data}}
```

I would suggest that maybe is better if we Base64 encode the `user_data` string here instead of expecting that is encoded by the caller.

```suggestion
          UserData: 
               Fn::Base64: {{user_data}}
```

---

> I would suggest that maybe is better if we Base64 encode the `user_data` string here instead of expecting that is encoded by the caller.

The problem with this approach is that, it is currently not easy to parse a multiline user string to the humilis-batch configuration. This probably mean the user will have to ensure they get the `newlines` and `spacing` right. But with this approach, a user can easily write their multiline shell script in a file and use a simple command such as `base64 -i config_init.service -o config_init_encode.service` to get the base64 string.